### ISSUE TITLE
Example Command Line Usage?

### ISSUE THREAD

**Issue by kexinhuang12345** [GIST-CSBL/DeepConv-DTI#1]
Hi, 

thanks for the code and detailed instruction. For the command line usage, could you Kindly provide an example input line since the brackets in the usage column seem not clear to me? Thanks!

**Comment by dlsrnsi** [comment]
Sorry for the late reply. I updated code, now it's compatible with python3.
For line example, if you have training dataset, `training_dti.csv`, `training_compound.csv` and `training_protein.csv` with right specification.
You can validate model with validation dataset, `validation_dti.csv`, `validation_compound.csv` and `validation_protein.csv` by using this command line.
`python DeepConvDTI.py ./training_dataset/training_dti.csv ./training_dataset/training_compound.csv ./training_dataset/training_protein.csv --validation -n validation_dataset -i ./validation_dataset/validation_dti.csv -d ./validation_dataset/validation_compound.csv -t ./validation_dataset/validation_protein.csv -W -c 512 128 -w 10 15 20 25 30 -p 128 -f 128 -r 0.0001 -n 30 -v Convolution -l 2500 -V morgan_fp_r2 -L 2048 -D 0 -a elu -F 128 -b 32 -y 0.0001 -o ./validation_output.csv -m ./model.model  -e 1`
This command will train model with given hyper-parameters for 1 epoch. (because of `-e 1`).
And resulting in validation result `validation_output.csv`, and corresponding model `model.model`

**Comment by kexinhuang12345** [comment]
Thanks for the update! One last question, if I also want to test the model selected from validation mode on a test set, what should I add? 

Because I saw there are two modes "validation" and "predict", if I find the best performing model on validation, how could I transfer it to test mode since there is no argument of input model? Thanks for the clarification!

**Comment by dlsrnsi** [comment]
Thank you for your question.
There are two ways to predict DTIs with command line
When you have `test_dti.csv`, `test_compound.csv` and `test_protein.csv`
`python DeepConvDTI.py ./training_dataset/training_dataset/0.csv ./training_dataset/merged_compound.csv ./training_dataset/merged_protein.csv --predict -n predict -i ./test_dataset/test_dti.csv -d ./test_dataset/test_compound.csv -t ./test_dataset/test_protein.csv  -c 512 128 -w 10 15 20 25 30 -p 128 -f 128 -r 0.0001 -n 30 -v Convolution -l 2500 -V morgan_fp_r2 -L 2048 -D 0 -a elu -F 128 -b 32 -y 0.0001 -o ./test_output.csv -m ./model.model -e 15 -W`
With this command, model will be trained with training dataset for 15 epochs and predict test dataset when it finished training, resulting in `test_output.csv` which have prediction score and its true label (because of `-W`)
The second way of prediction is using `predict_with_model.py`. If you have model which is saved from validation or something, you can use it. `python predict_with_model.py ./model.model  -n predict -i ./test_dataset/test_dti.csv -d ./test_dataset/test_compound.csv -t ./test_dataset/test_protein.csv -v Convolution -l 2500 -V morgan_fp_r2 -L 2048 -W -o test_result.csv`
this code will result in same result file with first command.
Hopefully, my explanation will help you.
`

**Comment by dlsrnsi** [comment]
In addition, you can evaluate the performances of prediction results with label by using `evaluate_performance.py`.
When you have an optimal threshold from validation and the names of the test dataset.
`python evaluate_performance.py test_result.csv -n predict -T 0.2` 
This command will report performances.

**Comment by kexinhuang12345** [comment]
Thank you for the detailed reply! I will try it out and send you some feedback!
